--- /usr/share/ansible/ansible-openstack-modules/neutron_network	2014-09-01 10:34:58.000000000 -0400
+++ neutron_network	2017-08-13 20:09:37.296183867 -0400
@@ -100,6 +100,11 @@ options:
         - Whether the state should be marked as up or down
      required: false
      default: true
+   port_security_enabled:
+     description:
+        - Whether the port security feature is enabled or not
+     required: false
+     default: true
 requirements: ["neutronclient", "keystoneclient"]
 
 '''
@@ -194,6 +199,7 @@ def _create_network(module, neutron):
         'router:external': module.params.get('router_external'),
         'shared': module.params.get('shared'),
         'admin_state_up': module.params.get('admin_state_up'),
+	 'port_security_enabled': module.params.get('enable_port_security'),
     }
 
     if module.params['provider_network_type'] == 'local':
@@ -245,7 +251,8 @@ def main():
             router_external = dict(default=False, type='bool'),
             shared = dict(default=False, type='bool'),
             admin_state_up = dict(default=True, type='bool'),
-            state = dict(default='present', choices=['absent', 'present'])
+            state = dict(default='present', choices=['absent', 'present']),
+	     port_security_enabled = dict(default=False, type='bool')
         ),
     )
 
