--- /usr/lib/python2.7/site-packages/shade/openstackcloud.py	2017-08-13 20:57:51.596886000 -0400
+++ openstackcloud.py	2017-08-13 20:09:37.297183873 -0400
@@ -3014,7 +3014,8 @@ class OpenStackCloud(
         return True
 
     def create_network(self, name, shared=False, admin_state_up=True,
-                       external=False, provider=None, project_id=None):
+                       external=False, port_security_enabled=True, 
+                       provider=None, project_id=None):
         """Create a network.
 
         :param string name: Name of the network being created.
@@ -3026,6 +3027,7 @@ class OpenStackCloud(
            { 'network_type': 'vlan', 'segmentation_id': 'vlan1' }
         :param string project_id: Specify the project ID this network
             will be created on (admin-only).
+	 :param bool port_security_enabled: Set port_security_enabled to true.
 
         :returns: The network object.
         :raises: OpenStackCloudException on operation error.
@@ -3033,6 +3035,7 @@ class OpenStackCloud(
         network = {
             'name': name,
             'admin_state_up': admin_state_up,
+	     'port_security_enabled': port_security_enabled,
         }
 
         if shared:
